### PR TITLE
[Issue #1657] Setup Next-Intl for App Pages

### LINKED ISSUE #1657: [Task]: Setup i18n with next-intl
### Summary

setup i18n with next-intl

### Acceptance criteria

Internationalization works on app router pages

### PR DESCRIPTION
**Pull request by acouch**
## Summary
Fixes #1657 

## Changes proposed
Enables next-intl to enable translation for app pages. This sets up next-intl and enables it for the search page and app not found page. This adds one other app component. There are three which will be removed for #1361 .

## Context for reviewers
The strings being declared and injected is still ugly and not addressed with this, but will be fixed with #1361 .

The `[locale]` folder has not been setup yet b/c it took over the full routing for the site. That folder will be setup in #1361 .

### CODE REVIEW

File: frontend/src/app/help/page.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+export async function generateMetadata({ params }: { params: RouteParams }) {
+  const t = await getTranslations({ locale: params.locale });
+  const meta: Metadata = {
+    title: t("Process.page_title"),
+  };
+  return meta;
+}
+
+export default async function Help() {
```
Comment (acouch) [inline frontend/src/app/help/page.tsx:35]: You use `useTransations()` and `useMessages()` if it is a sync function.

File: frontend/src/app/help/page.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import { Metadata } from "next";
```
Comment (acouch) [inline frontend/src/app/help/page.tsx:1]: This page was just for playing around, will delete.

**Comment by acouch** [comment]
This is green except for a single test for the `AppLayout.tsx`:

```
Error: `getTranslations` is not supported in Client Components.
    at /home/runner/work/simpler-grants-gov/simpler-grants-gov/frontend/node_modules/next-intl/dist/development/server/react-client/index.js:15:11
    at Layout (/home/runner/work/simpler-grants-gov/simpler-grants-gov/frontend/src/components/AppLayout.tsx:399:63)
    at renderWithHooks (/home/runner/work/simpler-grants-gov/simpler-grants-gov/frontend/node_modules/react-dom/cjs/react-dom.development.js:16305:18)
```

It is not a client component AFAICT, so not sure what the deal here is.

File: frontend/src/app/help/page.tsx (hunk 1 above)
Comment (rylew1) [reply frontend/src/app/help/page.tsx]: Believe all `use*` hooks should also only be used by client components.

**Comment by rylew1** [comment]
```
This is green except for a single test for the AppLayout.tsx:

Error: `getTranslations` is not supported in Client Components.
    at /home/runner/work/simpler-grants-gov/simpler-grants-gov/frontend/node_modules/next-intl/dist/development/server/react-client/index.js:15:11
    at Layout (/home/runner/work/simpler-grants-gov/simpler-grants-gov/frontend/src/components/AppLayout.tsx:399:63)
    at renderWithHooks (/home/runner/work/simpler-grants-gov/simpler-grants-gov/frontend/node_modules/react-dom/cjs/react-dom.development.js:16305:18)
It is not a client component AFAICT, so not sure what the deal here is.
```

I console logged in `AppLayout` and it only showed on the server. Wonder if the new `next-intl` changes are properly configured to support server components? 

I'm not fully capacity at the moment but I should be able to help debug this better on Thursday if it's still an issue.

File: frontend/src/components/AppBetaAlert.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { useTranslations } from "next-intl";
+
+import FullWidthAlert from "./FullWidthAlert";
+
+const BetaAlert = () => {
+  const t = useTranslations("Beta_alert");
+  const heading = t.rich("alert_title", {
```
Comment (acouch) [inline frontend/src/components/AppBetaAlert.tsx:7]: This file was added until #1361

File: frontend/src/components/AppLayout.tsx [hunk 4]
```diff
@@ -1,64 +1,59 @@
...
-  // locale?: string;
+  locale: string;
 };
 
-const Layout = ({ children }: Props) => {
-  // TODO: Remove during move to app router and next-intl upgrade
+export default function Layout({ children, locale }: Props) {
+  const t = useTranslations();
+
   const header_strings = {
```
Comment (acouch) [inline frontend/src/components/AppLayout.tsx:14]: These strings will be removed once we moved the app router and are no longer using both i18n and next-intl in https://github.com/HHS/simpler-grants-gov/issues/1361

File: frontend/src/components/AppBetaAlert.tsx (hunk 3 above)
Comment (rylew1) [reply frontend/src/components/AppBetaAlert.tsx:7]: Should we add a comment in the code that points to that ticket? Format we've been using is `// TODO (Issue #1361): ...`

**Comment by acouch** [comment]
Met with @rylew1 , want to look more closely at how the browser locale works and how to only allow the selected locales in the routing layer.

**Comment by acouch** [comment]
@rylew1 The PR does remove non language routes, I just had the `[locale]` folder in place locally w/o the rest of the config. The [browser locale docs](https://next-intl-docs.vercel.app/docs/routing/middleware#strategies) explain the browser config strategy. Think we are good to go.

**Review by rylew1 — APPROVED** [review]

**Pull request merged**